spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -146,7 +149,9 @@ public TaskLifecycleListener(TaskRepository taskRepository,
 		this.taskExplorer = taskExplorer;
 		this.taskProperties = taskProperties;
 		this.taskListenerExecutorObjectFactory = taskListenerExecutorObjectFactory;
-		this.taskMetrics = new TaskMetrics();
+		taskMetrics = meterRegistry != null ?
+			new TaskMetrics(meterRegistry) : null;
+		this.taskMetrics = taskMetrics;
```

Comment:
Wouldn't this be the same as:
```
this.taskMetrics = meterRegistry != null ? new TaskMetrics(meterRegistry) : null;
```
so that we can also remove `TaskMetrics taskMetrics;` from line\#138

Reply:
Last time I listen to the recommendation from Intellij :-D